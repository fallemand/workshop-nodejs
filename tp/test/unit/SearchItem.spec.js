const React = require('react');
const renderer = require('react-test-renderer');
const SearchItem = require('../../views/components/SearchItem');
const SearchMock = require('../../mocks/search.json');

describe("SearchItem (snapshot)", () => {
  it("should render an item correctly when item data is provided", () => {
    const item = renderer.create(<SearchItem {...SearchMock.results[0]} />).toJSON();
    expect(item).toMatchSnapshot();
  })
});
