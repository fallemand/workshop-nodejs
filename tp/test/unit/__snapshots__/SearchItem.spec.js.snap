// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem (snapshot) should render an item correctly when item data is provided 1`] = `
<a
  className="search-item"
  href="/app/items/MLA622544573"
>
  <div
    className="search-item__picture"
  >
    <img
      alt="Papa Bastón Congelada - $ 32 /kg - Sin Aceite Agregado"
      src="http://mla-s2-p.mlstatic.com/855221-MLA20747286710_062016-I.jpg"
    />
  </div>
  <div
    className="search-item__description"
  >
    <span
      className="search-item__price"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"currency-symbol\\">A</span><span class=\\"amount\\">RS80</span><sup class=\\"fractional\\">00</sup>",
        }
      }
    />
    <p
      className="search-item__title"
    >
      Papa Bastón Congelada - $ 32 /kg - Sin Aceite Agregado
    </p>
  </div>
  <div
    className="search-item__location"
  />
</a>
`;
